qt/scientific_interfaces/Indirect/Iqt.cpp

```diff
@@ -31,14 +34,71 @@ std::size_t getWsNumberOfSpectra(std::string const &workspaceName) {
   return getADSMatrixWorkspace(workspaceName)->getNumberHistograms();
 }
 
-bool checkADSForWorkspace(std::string const &workspaceName) {
+bool doesExistInADS(std::string const &workspaceName) {
   return AnalysisDataService::Instance().doesExist(workspaceName);
 }
 
 bool isWorkspacePlottable(MatrixWorkspace_sptr workspace) {
   return workspace->y(0).size() > 1;
 }
 
+std::string
+checkInstrumentParametersMatch(Instrument_const_sptr sampleInstrument,
+                               Instrument_const_sptr resolutionInstrument,
+                               std::string const &parameter) {
+  if (!sampleInstrument->hasParameter(parameter))
+    return "Could not find the " + parameter + " for the sample workspace.";
+  if (!resolutionInstrument->hasParameter(parameter))
+    return "Could not find the " + parameter +
+           " for the resolution workspaces.";
+  if (sampleInstrument->getStringParameter(parameter)[0] !=
+      resolutionInstrument->getStringParameter(parameter)[0])
+    return "The sample and resolution must have matching " + parameter + "s.";
+  return "";
+}
+
+std::string checkParametersMatch(MatrixWorkspace_const_sptr sampleWorkspace,
+                                 MatrixWorkspace_const_sptr resolutionWorkspace,
+                                 std::string const &parameter) {
+  auto const sampleInstrument = sampleWorkspace->getInstrument();
+  auto const resolutionInstrument = resolutionWorkspace->getInstrument();
+  return checkInstrumentParametersMatch(sampleInstrument, resolutionInstrument,
+                                        parameter);
+}
+
+std::string checkParamsMatch(std::string const &sampleName,
```

Comment:
I would probably overload `checkParametersMatch`, rather than using slightly different function names. That way you don't have to remember which function name is for which inputs